### PR TITLE
ZK-4810: Radiogroup should offer setDisabled(..)/isDisabled(..), esp.…

### PR DESCRIPTION
**Pull request by plijyfes**
… when used with model & renderer

### CODE REVIEW

File: zktest/src/archive/test2/F96-ZK-4810-2.zul [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+			</radiogroup>
+			<button label="set another model" onClick="rg2.setModel(list2)" />
+			<button label="add" onClick='rg2.getModel().add("new1")' />
+			<button label="remove" onClick='rg2.getModel().remove(rg2.getModel().size()-1)' />
+			<button label="toggle radiogroup2 disabled" onClick="rg2.setDisabled(!rg2.isDisabled())" />
+		</hlayout>
+		<label value="Test Renderer" />
+		<label multiline="true">
+			1. you shall see the radios in radiogroup3 are not disabled.
+			2. selecet the "renderer[radio2]", and click the "toggle radiogroup3 disabled" button.
```
Comment (plijyfes) [inline zktest/src/archive/test2/F96-ZK-4810-2.zul:69]: typo

File: zktest/src/archive/test2/F96-ZK-4810.zul [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		History:
+				Thu Jul 15 10:21:46 CST 2021, Created by leon
+
+Copyright (C) 2021 Potix Corporation. All Rights Reserved.
+
+-->
+<zk>
+	<label multiline="true">
+		1. you shall see the radios in radiogroup(1~4) are not disabled.
+		2. selecet the "radio2", and click the "toggle radiogroup disabled" button.
```
Comment (plijyfes) [inline zktest/src/archive/test2/F96-ZK-4810.zul:18]: typo

File: zktest/src/archive/test2/config.properties [hunk 3]
```diff
@@ -3040,6 +3040,8 @@ B90-ZK-4431.zul=A,E,Multislider
 ##manually##B96-ZK-4953.zul=A,E,Embedded,Include,defer
 ##zats##B96-ZK-4949.zul=A,E,VParent,unbind
 ##zats##B96-ZK-4868.zul=A,M,signature,backgroundColor
+##zats##B96-ZK-4810.zul=A,M,Radiogroup,disabled
+##zats##B96-ZK-4810-2.zul=A,M,Radiogroup,disabled,model
```
Comment (plijyfes) [inline zktest/src/archive/test2/config.properties:3044]: feature

File: zul/src/org/zkoss/zul/Radiogroup.java [hunk 4]
```diff
@@ -362,6 +366,31 @@ private static int fixSelIndex(Component comp, MutableInteger cur) {
...
+	 */
+	public boolean isDisabled() {
+		return _disabled;
+	}
+	
+	/** Sets whether it is disabled.
+	 * All the radios belong to this radiogroup will be set to disabled or not disabled together.
+	 * Notice: Once a radio is added to a disabled radiogroup (including external radio),
+	 * the radio will be set to disabled too.
+	 * @since 9.6.0
```
Comment (plijyfes) [inline zul/src/org/zkoss/zul/Radiogroup.java:382]: @param

File: zul/src/org/zkoss/zul/Radiogroup.java [hunk 5]
```diff
@@ -737,4 +766,22 @@ public void onPageDetached(Page page) {
...
+	public void onChildAdded(Component child) {
+		super.onChildAdded(child);
+		if (_disabled) {
+			if (child instanceof Radio) {
+				((Radio) child).setDisabled(true);
+			} else {
+				// ZK-4810: find all nested radio
+				List<Radio> items = new ArrayList<Radio>();
+				getItems0(child, items);
+				for (Iterator<Radio> it = items.iterator(); it.hasNext();) {
```
Comment (plijyfes) [inline zul/src/org/zkoss/zul/Radiogroup.java:780]: use for (bla : bla)

**Review by scribetw — CHANGES_REQUESTED** [review]
need to be refined

**Review by DevChu — APPROVED** [review]

**Review by scribetw — APPROVED** [review]

**Pull request merged**